### ISSUE TITLE
[Bug]: elpaca-update-all is failing on some package (due to API  limits @ GitHub?)

### ISSUE THREAD

**Issue by nrvale0** [progfolio/elpaca#156]
### Elpaca Version

```
"Elpaca 084a7df HEAD -> master, origin/master, origin/HEAD
installer:      0.4
emacs-version:  GNU Emacs 28.2 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.37, cairo version 1.16.0)
git --version:  git version 2.34.1"```


### Operating System

Ubuntu 22.04

### Description

My Emacs config leverages about 160 packages. If I do a `elpaca-update-all` to get the latest code for all of those packages the last 40 or so packages will normally fail to update with a message something like:

```
error: Could not fetch origin
```

or

```
fatal: unable to access '<url to repo>': Failed to connec...
```

As most of the modules are being pulled from GitHub I suspect this might be due to some sort of GH API rate-limit but I'm not sure how I might verify. They are all https URLs so I cannot easily sniff the traffic to verify.

1) Perhaps you have an idea how I can verify my suspicion?

2) If I'm correct, perhaps there's some ability to tweak the amount of ||ism in the elpaca code when it comes to repo updates/API interactions?

**Comment by progfolio** [comment]
Thanks for taking the time to fill out a bug report.

> My Emacs config leverages about 160 packages. If I do a `elpaca-update-all` to get the latest code for all of those packages the last 40 or so packages will normally fail to update with a message something like:
> 
> ```
> error: Could not fetch origin
> ```
> 
> or
> 
> ``` fatal: unable to access '<url to repo>': Failed to connec... ```
> 
> As most of the modules are being pulled from GitHub I suspect this might be due
> to some sort of GH API rate-limit but I'm not sure how I might verify. They are
> all https URLs so I cannot easily sniff the traffic to verify.
> 
> 1) Perhaps you have an idea how I can verify my suspicion?

Elpaca is using git directly under the hood.
When things start failing to connect, you could open a terminal and try to manually `git fetch --all -v` or whatever command you want to test with (you can see the exact commands which were run by evaluating `(elpaca-log "#(verbosity) \\\\$")`).
I would be surprised if 160 requests was considered too many, but I'm not sure if Github throttles by IP in that way. Let me know what you find out.

Are you using anything like a proxy?
  
> 2) If I'm correct, perhaps there's some ability to tweak the amount of ||ism in
> the elpaca code when it comes to repo updates/API interactions?

The `elpaca-queue-limit` user option can be set to limit the max number of orders processed at once. e.g. `(setq elpaca-queue-limit 30)`.

**Comment by nrvale0** [comment]
I set =elpaca-queue-limit= to =10= and that seems to have resolved the issues with =elpaca-fetch-all=. I can only surmise that it then flys under the GH API limit radar. 

I tried to get a better understanding of the situation but the GH API limit docs + licensing terms are annoyingly complex and take into account free vs paid, logged-in versus not logged in, GHE versus not. Yuck. 

Thx for your help.
